app/controllers/user_controller.php

```diff
@@ -0,0 +1,106 @@
+<?php
+class UserController extends AppController
+{
+    public function register()
+    {
+        $register_info = new RegisterInfo;
+        $user = new User;
+        $page = Param::get('page_next', 'register');
+        $error = false;
+
+        switch ($page) {
+            case 'register':
+                break;
+            case 'write_end':
+                $register_info->firstname = Param::get('firstname');
+                $register_info->lastname = Param::get('lastname');
+                $register_info->email = Param::get('email');
+                $register_info->username = Param::get('username');
+                $register_info->password = Param::get('password');
+                $register_info->validate_password = Param::get('validate_password');
```

Comment:
For L15-20 po, there is no need to create a new model just for validation. You could remove this and just validate the information passed at the User model :bow: